src/wrappers/themis/java/com/cossacklabs/themis/SecureCell.java

```diff
@@ -16,18 +16,529 @@
+         * @throws RuntimeException on internal encryption failure
+         *         (with {@link SecureCellException} as its cause)
+         */
+        @NotNull
+        @Contract(pure = true)
+        SecureCellData encrypt(byte[] data);
+
+        /**
+         * Decrypts the provided message with associated context.
+         * <p>
+         * Secure Cell validates association with the context data, decrypts the message,
+         * and verifies its integrity using the provided authentication token.
+         * <p>
+         * You need to provide the same context data as provided to {@link #encrypt(byte[], byte[])}.
+         * You can also decrypt data encrypted with {@link #encrypt(byte[])} by using an empty context.
+         *
+         * @param data      encrypted data, cannot be empty
+         * @param token     authentication token, cannot be empty
+         * @param context   associated context, may be empty or {@code null}
+         *
+         * @return decrypted data if everything goes well
+         * @throws SecureCellException if decryption failed: for example,
+         *         if the data has been tampered with, if the secret is incorrect,
+         *         or if the associated context does not match the one used to encrypt the data
+         * @throws InvalidArgumentException if data or token is empty
+         */
+        @NotNull
+        @Contract(pure = true)
+        byte[] decrypt(byte[] data, byte[] token, @Nullable  byte[] context) throws SecureCellException;
+
+        /**
+         * Decrypts the provided message with associated context.
+         * <p>
+         * Secure Cell decrypts the message and verifies its integrity
+         * using the provided authentication token.
+         * <p>
+         * Use this method to decrypt data encrypted with {@link #encrypt(byte[])}.
+         * If you use associated context, call {@link #decrypt(byte[], byte[], byte[])} instead.
+         *
+         * @param data  encrypted data, cannot be empty
+         * @param token authentication token, cannot be empty
+         *
+         * @return decrypted data if everything goes well
+         * @throws SecureCellException if decryption failed: for example,
+         *         if the data has been tampered with, if the secret is incorrect,
+         *         or if the associated context does not match the one used to encrypt the data
+         * @throws InvalidArgumentException if data or token is empty
+         */
+        @NotNull
+        @Contract(pure = true)
+        byte[] decrypt(byte[] data, byte[] token) throws SecureCellException;
+    }
+
+    /**
+     * Makes a new Secure Cell in Token Protect mode secured by symmetric key.
+     *
+     * @param key symmetric key to use
+     *
+     * @return a new {@link SecureCell.TokenProtect} instance
+     *
+     * @since JavaThemis 0.13
+     */
+    @NotNull
+    @Contract(value = "_ -> new", pure = true)
```

Comment:
oh... java doesn't provide some named constants for pre-defined values for contracts? to avoid using hardcoded strings

Reply:
> java doesn't provide some named constants for pre-defined values for contracts?

[Nope](https://www.jetbrains.com/help/idea/contract-annotations.html), consider it source code.